cpp/cograph_rec/cograph/Coloring.cpp

```diff
@@ -0,0 +1,84 @@
+#include "cograph_rec/Cotree.hpp"
+
+namespace Koala {
+    void Cotree::SubtreeColors(long long v) {
```

Comment:
Maybe it's better to write CographColoring class with run() method invoking (1) building cotree, (2) running proivate method recurse_color(CoTree &T, long long v) inside, with exactly the same functionality as this method.

PS. Look at the Coloring classes to see what is the input/output interface i.e. ther should be getColoring method etc.